diff --git a/CMakeLists.txt b/CMakeLists.txt
index 19fdfa46..9727c5a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1056,6 +1056,7 @@ elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64" OR CMAKE_GENERATOR_PLATFORM_LW
         endif()
         if (LLAMA_AVX512_VNNI)
             list(APPEND ARCH_FLAGS -mavx512vnni)
+            list(APPEND ARCH_FLAGS -mavx512vl)
         endif()
     endif()
 elseif (${CMAKE_SYSTEM_PROCESSOR} MATCHES "ppc64")
